packages/compass/src/main/window-manager.ts

```diff
@@ -77,11 +78,59 @@ const DEFAULT_URL =
   process.env.COMPASS_INDEX_RENDERER_URL ||
   pathToFileURL(path.join(__dirname, 'index.html')).toString();
 
-async function showWindowWhenReady(bw: BrowserWindow) {
+async function showWindowWhenReady(bw: BrowserWindow, isMaximized?: boolean) {
   await once(bw, 'ready-to-show');
+  if (isMaximized) {
+    // win.maximize() maximizes the window.
+    // This will also show (but not focus) the window if it isn't being displayed already.
+    bw.maximize();
+    bw.focus();
+    return;
+  }
+
   bw.show();
 }
 
+/**
+ * Save window bounds to preferences
+ */
+async function saveWindowBounds(
+  window: BrowserWindow,
+  compassApp: typeof CompassApplication
+) {
+  try {
+    const bounds = window.getBounds();
+    const isMaximized = window.isMaximized();
+
+    await compassApp.preferences.savePreferences({
+      windowBounds: {
+        x: bounds.x,
+        y: bounds.y,
+        width: bounds.width,
+        height: bounds.height,
+        isMaximized,
+      },
+    });
+  } catch (err) {
+    // eslint-disable-next-line no-console
+    console.error(err);
```

Comment:
We don't check the console for debugging often. If a user ends up in a situation where they're unable to save/restore their window we'll typically start by asking them for their Compass log file: https://www.mongodb.com/docs/compass/troubleshooting/logs/#view-compass-logs 

Something like:

```suggestion
    log.warn(
      mongoLogId(1_001_000_TODO), // The next available `mongoLogId` can be found with `npm run check-logids`.
      'Window Manager',
      'Failed to save window bounds',
      {
        message: (err as Error).message,
      }
    );```

Reply:
Updated with the IDs generated by npm `run check-logids`